### PR TITLE
Implement high and low number search with tests

### PR DESCRIPTION
**Pull request by luqp**

### CODE REVIEW

File: src/main/java/org/fundacionjala/lucero/Kata.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    public static String highAndLow(final String data) {
+        if (data.isEmpty() || data.length() == 1) {
+            return data;
+        }
+
+        List<Integer> numbers = recoverNumbers(data);
+
+        Collections.sort(numbers);
```
Comment (marcos-x86) [inline src/main/java/org/fundacionjala/lucero/Kata.java]: Sorting operations are too expensive to compute with large collections, so please consider using `Collections.max()` and `Collections.min()` utility methods.

File: src/main/java/org/fundacionjala/lucero/Kata.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    private Kata() { }
+
+    public static List<Integer> recoverNumbers(final String data) {
+        Stream<Integer> numbers = Arrays.stream(data.split(" "))
+                .map(Integer::parseInt);
+        return numbers.collect(Collectors.toList());
+    }
+
+    public static String highAndLow(final String data) {
+        if (data.isEmpty() || data.length() == 1) {
```
Comment (marcos-x86) [inline src/main/java/org/fundacionjala/lucero/Kata.java]: Please consider simplifying this boolean expression. Something like this:
```java
if (data.length() <= 1) {
  // implementation...
}
```

File: src/main/java/org/fundacionjala/lucero/Kata.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        }
+
+        List<Integer> numbers = recoverNumbers(data);
+
+        Collections.sort(numbers);
+        return buildResult(numbers.get(numbers.size() - 1), numbers.get(0));
+    }
+
+    private static String buildResult(final Integer high, final Integer low) {
+        return high.toString() + " " + low.toString();
```
Comment (marcos-x86) [inline src/main/java/org/fundacionjala/lucero/Kata.java]: Please consider avoid using `+` operator to concatenate Strings. You can use `String.join` utility method.

**Review by marcos-x86 — CHANGES_REQUESTED** [review]

File: src/main/java/org/fundacionjala/lucero/Kata.java (hunk 3 above)
Comment (luqp) [reply src/main/java/org/fundacionjala/lucero/Kata.java]: Hi Marcos, I already improved my code, thanks for your reviews.

**Review by marcos-x86 — APPROVED** [review]
LGTM!

**Pull request merged**